src/Cheats.c

```diff
@@ -928,40 +928,40 @@ int cheatsCheckKeys(u32 keys, u32 extended)
 		}
         break;
       case GSA_8_BIT_POINTER :
-        if ((CPUReadMemory(cheatsList[i].address)>=0x02000000) && (CPUReadMemory(cheatsList[i].address)<0x02040000) ||
-            (CPUReadMemory(cheatsList[i].address)>=0x03000000) && (CPUReadMemory(cheatsList[i].address)<0x03008000))
+        if ((CPUReadMemory(cheatsList[i].address)>=0x02000000 && CPUReadMemory(cheatsList[i].address)<0x02040000) ||
+            (CPUReadMemory(cheatsList[i].address)>=0x03000000 && CPUReadMemory(cheatsList[i].address)<0x03008000))
         {
           CPUWriteByte(CPUReadMemory(cheatsList[i].address)+((cheatsList[i].value & 0xFFFFFF00) >> 8),
                        cheatsList[i].value & 0xFF);
         }
         break;
       case GSA_16_BIT_POINTER :
-        if ((CPUReadMemory(cheatsList[i].address)>=0x02000000) && (CPUReadMemory(cheatsList[i].address)<0x02040000) ||
-            (CPUReadMemory(cheatsList[i].address)>=0x03000000) && (CPUReadMemory(cheatsList[i].address)<0x03008000))
+        if ((CPUReadMemory(cheatsList[i].address)>=0x02000000 && CPUReadMemory(cheatsList[i].address)<0x02040000) ||
+            (CPUReadMemory(cheatsList[i].address)>=0x03000000 && CPUReadMemory(cheatsList[i].address)<0x03008000))
         {
           CPUWriteHalfWord(CPUReadMemory(cheatsList[i].address)+((cheatsList[i].value & 0xFFFF0000) >> 15),
                        cheatsList[i].value & 0xFFFF);
         }
         break;
       case GSA_32_BIT_POINTER :
-        if ((CPUReadMemory(cheatsList[i].address)>=0x02000000) && (CPUReadMemory(cheatsList[i].address)<0x02040000) ||
-            (CPUReadMemory(cheatsList[i].address)>=0x03000000) && (CPUReadMemory(cheatsList[i].address)<0x03008000))
+        if ((CPUReadMemory(cheatsList[i].address)>=0x02000000 && CPUReadMemory(cheatsList[i].address)<0x02040000) ||
+            (CPUReadMemory(cheatsList[i].address)>=0x03000000 && CPUReadMemory(cheatsList[i].address)<0x03008000))
         {
           CPUWriteMemory(CPUReadMemory(cheatsList[i].address),
                        cheatsList[i].value);
         }
         break;
       case GSA_8_BIT_ADD :
         CPUWriteByte(cheatsList[i].address,
-                    (cheatsList[i].value & 0xFF) + CPUReadMemory(cheatsList[i].address) & 0xFF);
+                    (cheatsList[i].value & 0xFF) + (CPUReadMemory(cheatsList[i].address) & 0xFF));
         break;
       case GSA_16_BIT_ADD :
         CPUWriteHalfWord(cheatsList[i].address,
-                        (cheatsList[i].value & 0xFFFF) + CPUReadMemory(cheatsList[i].address) & 0xFFFF);
+                        (cheatsList[i].value & 0xFFFF) + (CPUReadMemory(cheatsList[i].address) & 0xFFFF));
         break;
       case GSA_32_BIT_ADD :
         CPUWriteMemory(cheatsList[i].address ,
-                       cheatsList[i].value + CPUReadMemory(cheatsList[i].address) & 0xFFFFFFFF);
+                       cheatsList[i].value + (CPUReadMemory(cheatsList[i].address) & 0xFFFFFFFF));
```

Comment:
operator precedence in C is tricky. can you do a before/after comparison of the compiled object files?